### PR TITLE
added coach controller

### PR DESCRIPTION
**Pull request by chukulert**
I am not very sure this is right. I tried to follow what rail routes generated.

Routes for the coach namespace (from rail routes):

GET /coach/lessons(.:format)    coach/lessons#index
POST /coach/lessons(.:format)       coach/lessons#create
GET /coach/lessons/new(.:format)     coach/lessons#new
GET /coach/lessons/:id/edit(.:format)      coach/lessons#edit
GET /coach/lessons/:id(.:format)          coach/lessons#show
PATCH /coach/lessons/:id(.:format)       coach/lessons#update
PUT /coach/lessons/:id(.:format)        coach/lessons#update
DELETE /coach/lessons/:id(.:format)

### CODE REVIEW

File: app/controllers/coach/lessons_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    def destroy
+      @lesson.destroy
+      redirect_to action: 'index'
+    end
+
+    private
+
+    def lesson_params
+      params.require(:lesson).permit(:start_time, :end_time, :location, :price)
```
Comment (NAVentilateN) [inline app/controllers/coach/lessons_controller.rb:42]: rename to start_date_time and end_date_time to align with model

File: app/controllers/coach/lessons_controller.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def new
+      # @user = current_user
+      @lesson = Lesson.new
+    end
+
+    def create
+      @lesson = Lesson.new(lesson_params)
+      # @user = current_user
+      @lesson.coach_id = current_user
```
Comment (NAVentilateN) [inline app/controllers/coach/lessons_controller.rb:17]: is this line required? Current;y we don't really need to verify the coach.

File: app/controllers/coach/lessons_controller.rb (hunk 1 above)
Comment (chukulert) [reply app/controllers/coach/lessons_controller.rb]: Done!

File: app/controllers/coach/lessons_controller.rb (hunk 2 above)
Comment (chukulert) [reply app/controllers/coach/lessons_controller.rb:17]: Hm i wanted to add current user as the coach_id since it is a required attribute before saving

Comment (NAVentilateN) [reply app/controllers/coach/lessons_controller.rb:17]: Oh right! sorry about it!

File: app/controllers/coach/lessons_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      @lesson.status = false
```
Comment (NAVentilateN) [inline app/controllers/coach/lessons_controller.rb:18]: fine for now. many in the model we can add default as false

**Pull request merged**